src/SharedMoudule/Components/Filter/Filter.tsx

```diff
@@ -0,0 +1,98 @@
+import axios from "axios";
+import styles from "./Filter.module.css";
+import { baseUrl } from "../../../Constants/Components/Urls";
+import { useEffect, useState } from "react";
+
+export default function Filter({ setUsersList }) {
+  const [searchByName, setSearchByName] = useState("");
+  const [searchByEmail, setSearchByEmail] = useState("");
+  const [searchByCountry, setSearchByCountry] = useState("");
+  const [searchByGroup, setSearchByGroup] = useState(1||2);
+  const [selectAnyThing, setSelectAnyThing] = useState(null);
+  const [params, setParams] = useState("");
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      callFiltrationUser();
+    }, 500);
+    return () => {
+      clearTimeout(timer);
+      setSearchByName("");
+      setSearchByEmail("");
+      setSearchByGroup("");
+      setSearchByCountry("");
+    };
+  }, [params]);
+
+  async function callFiltrationUser() {
+    try {
+      const { data } = await axios.get(
+        `${baseUrl}/Users/?userName=${searchByName}&email=${searchByEmail}&country=${searchByCountry}&groups=${searchByGroup}&pageSize=10&pageNumber=1`,
```

Comment:
I suggest passing  the params via the params object instead of writing them manually in the URL